### DISCUSSION TITLE
Styling

### DISCUSSION THREAD

**Discussion by kevintupper** [mstaal/msal_streamlit_authentication#1] [Ideas]
There is no way to control the styling.  It's a big button and all I can control is the text on it. Any chance it can be written to expose the styles?

**Comment by mstaal** [comment]
I think it's a good suggestion. Will try to implement that one of the coming days. I imagine something like exposing the style, class or id attribute so that you are able to set the styles via multiple means.

**Comment by mstaal** [comment]
> There is no way to control the styling. It's a big button and all I can control is the text on it. Any chance it can be written to expose the styles?

Hi @kevintupper, I just made a new version (v1.0.8) which should add support for supplying HTML class or HTML id to the button element. The README shows the syntax on how to do that. For now, this should allow you to set the styling by letting the button inherit from whatever styling have been put onto those. But I have not tested it thoroughly, so feel free to give it a shot. I wanted to support the `style` attribute as well, but it seems like React wants for the contents of that attribute to adhere to a certain format. I hope the additon is of use to you.

**Comment by kwaksaewon** [comment]
Hey @mstaal , when you have time could you please upload a sample of how someone can dynamically change the button style by id? 

I've been trying to do it by doing this:
`with open('style.css') as f:
            st.markdown(f"<style>{f.read()}</style>", unsafe_allow_html = True)`

and then initializing `msal_authentication(
            .
            .
            .
            html_id="login_and_out_button",
            key=1)`, with the style.css containing this:

`#login_and_out_button {
    background-color: azure
}`,

but when I inspect the page, the background-color of the login_and_out_button html id still is this: `background-color: inherit`

**Comment by idlaneev** [comment]
Hey , guys , is there any updates on issue?
